src/groups/general_linear.jl

```diff
@@ -0,0 +1,238 @@
+    end
+    mpv = check_tangent_vector(Euclidean(n, n; field = 𝔽), p, X; kwargs...)
+    mpv === nothing || return mpv
+    return nothing
+end
+
+decorator_transparent_dispatch(::typeof(exp), ::GeneralLinear, args...) = Val(:parent)
+decorator_transparent_dispatch(::typeof(retract!), ::GeneralLinear, args...) = Val(:parent)
+decorator_transparent_dispatch(::typeof(log), ::GeneralLinear, args...) = Val(:parent)
+
+function default_metric_dispatch(
+    ::GeneralLinear{n,ℝ},
+    ::LeftInvariantMetric{EuclideanMetric},
+) where {n}
+    return Val(true)
+end
+
+function exp!(::GeneralLinear, q, p, X)
+    mul!(q, exp(X)', exp(X - X'))
+    return copyto!(q, p * q)
+end
+function exp!(::GeneralLinear{1}, q, p, X)
+    p1 = p isa Identity ? p : p[1]
+    q[1] = p1 * exp(X[1])
+    return q
+end
+function exp!(::GeneralLinear{2}, q, p, X)
+    A = exp_2x2(X)'
+    B = similar(X)
+    B .= X .- X'
+    exp_2x2!(B, B)
+    mul!(q, A, B)
+    return copyto!(q, p * q)
+end
+function exp!(
+    M::MetricManifold{𝔽,<:GeneralLinear{n,𝔽},<:LeftGLInvariantMetric},
+    q,
+    p,
+    X,
+) where {n,𝔽}
+    g = metric(M)
+    T = eltype(X)
+    ω = T(g.μc / g.μ)
+    α, β = (1 - ω) / 2, (1 + ω) / 2
+    mul!(q, exp(α .* X .+ β .* X'), exp(β .* (X .- X')))
+    copyto!(q, p * q)
+    return q
+end
+
+flat!(::GeneralLinear, ξ::CoTFVector, p, X::TFVector) = copyto!(ξ, X)
+
+get_coordinates(::GeneralLinear{n,ℝ}, p, X, ::DefaultOrthonormalBasis) where {n} = vec(X)
+
+function get_coordinates!(::GeneralLinear{n,ℝ}, Xⁱ, p, X, ::DefaultOrthonormalBasis) where {n}
+    return copyto!(Xⁱ, X)
+end
+
+function get_vector(::GeneralLinear{n,ℝ}, p, Xⁱ, ::DefaultOrthonormalBasis) where {n}
+    return reshape(Xⁱ, n, n)
+end
+
+function get_vector!(::GeneralLinear{n,ℝ}, X, p, Xⁱ, ::DefaultOrthonormalBasis) where {n}
+    return copyto!(X, Xⁱ)
+    return X
+end
+
+function group_exp!(::GeneralLinear{1}, q, X)
+    q[1] = exp(X[1])
+    return q
+end
+group_exp!(::GeneralLinear{2}, q, X) = exp_2x2!(q, X)
+
+function group_log!(::GeneralLinear{1}, X, p)
+    X[1] = log(p[1])
+    return X
+end
+
+inner(::GeneralLinear, X, Y) = dot(X, Y)
+function inner(
+    M::MetricManifold{𝔽,<:GeneralLinear{n,𝔽},<:LeftGLInvariantMetric},
+    p,
+    X,
+    Y,
+) where {n,𝔽}
+    g = metric(M)
+    return (
+        (g.μ + g.μc) * dot(X, Y) / 2 +
+        (g.μ - g.μc) * dot(X, Y') / 2 +
+        (g.κ - g.μ) * tr(X) * tr(Y) / n
+    )
+end
+
+invariant_metric_dispatch(::GeneralLinear, ::LeftAction) = Val(true)
+
+inverse_translate(::GeneralLinear, p, q, ::LeftAction) = p \ q
+inverse_translate(::GeneralLinear, p, q, ::RightAction) = q / p
+
+inverse_translate_diff(::GeneralLinear, p, q, X, ::LeftAction) = X
+inverse_translate_diff(::GeneralLinear, p, q, X, ::RightAction) = p * X / p
+
+function inverse_translate_diff!(G::GeneralLinear, Y, p, q, X, conv::ActionDirection)
+    return copyto!(Y, inverse_translate_diff(G, p, q, X, conv))
+end
+
+function log!(G::GeneralLinear, X, p, q)
```

Comment:
Maybe it would make sense to write a more generic `inverse_retraction` with an `GeneralLinearInverseRetraction` type that allows specifying parameters for that nonlinear solver? Or does the solver more or less works well enough with these default parameters?

Reply:
One could even do this more generally by using a `ApproximateLogarithmicRetraction` type for these parameters?

Reply:
Yes, that's right. And then it could also be used for example in `Stiefel`.

Reply:
Exactly. We could adapt the variant from GeomStats using the same type. For generality maybe with a dictionary of options for the solver? Something like that.

Reply:
> Exactly. We could adapt the variant from GeomStats using the same type. For generality maybe with a dictionary of options for the solver? Something like that.

Adapt what variant? GeomStats' `log` for general linear is equivalent to our `GroupLogarithmicInverseRetraction `.

Reply:
I think maybe an `ApproximateInverseRetraction{<:AbstractRetractionMethod}` (or maybe `SolvedInverseRetraction`?) would be useful, and then, yes, we could have the kwargs be forwarded to the solver.

Reply:
Oh, I was referring to the approximate (numerical optimization) log of GeomStats `Stiefel`, sorry for not being precise.

Ah a typed ApproximateRetraction would be a better idea, yes.